Creating initial documentation for the quality assurance process on the platform

pjhill opened the pull request:
This documentation describes a process for VFS teams to create a lightweight quality assurance practice for use on the platform.

Assumptions:
1. The QA process describes providing output at the end of a QA effort to the VSP Product Support team for review.
2. VFS teams are expected to use the supplied templates for various deliverables.
3. A test cases management software integration will be available on the platform in the future.

pjhill committed:
Add files via upload

pjhill committed:
Updating links from README

pjhill committed:
Updating link to process diagram image

pjhill committed:
Update process.md

pjhill committed:
Update deliverables.md

pjhill committed:
Update process.md

- Link to LucidChart
- Permissions to edit LucidChart document granted

pjhill committed:
Update process.md

Fixed link

pjhill committed:
Update process.md

Updated appearance

pjhill committed:
Update process.md

pjhill committed:
Update process.md

pjhill committed:
Update deliverables.md

pjhill committed:
Update deliverables.md

pjhill committed:
Update deliverables.md

pjhill committed:
Update deliverables.md

pjhill committed:
Create create-an-issue

pjhill committed:
Update README.md

pjhill committed:
Rename create-an-issue to create-an-issue.md

pjhill committed:
Update create-an-issue.md

pjhill committed:
Update create-an-issue.md

pjhill committed:
Update process.md

pjhill committed:
Update deliverables.md

pjhill committed:
Update deliverables.md

pjhill committed:
Update deliverables.md

pjhill committed:
Update README.md

pjhill committed:
Update process.md

pjhill committed:
Update process.md

pjhill committed:
Update process.md

pjhill committed:
Update README.md

pjhill committed:
Create / link to material for load testing

pjhill committed:
Update README.md

pjhill committed:
Update README.md

pjhill committed:
Create auth-session-loadtest.md

pjhill committed:
Update README.md

pjhill committed:
Update auth-session-loadtest.md

pjhill committed:
Add files via upload

pjhill committed:
Update auth-session-loadtest.md

pjhill committed:
Update auth-session-loadtest.md

pjhill committed:
Update auth-session-loadtest.md

pjhill committed:
Add files via upload

pjhill committed:
Update auth-session-loadtest.md

pjhill committed:
Update README.md

pjhill committed:
Create README.md

pjhill committed:
Add files via upload

moving images to /images/ folder

pjhill committed:
Delete jenkins-build-with-parameters.png

pjhill committed:
Delete jenkins-build.png

pjhill committed:
Delete jenkins-click-the-build.png

pjhill committed:
Delete jenkins-console-output.png

pjhill committed:
Delete QA-Process-Flow.png

pjhill committed:
Add files via upload

pjhill committed:
Delete jenkins-session-cookie-output.png

pjhill committed:
Delete jenkins-trimmed-session-cookies.PNG

pjhill committed:
Delete jenkins-login.png

pjhill committed:
Delete jenkins-trimmed-session-cookies.PNG

pjhill committed:
Add files via upload

pjhill committed:
Update auth-session-loadtest.md

pjhill committed:
Update process.md

pjhill committed:
Update process.md

pjhill committed:
Update auth-session-loadtest.md

peterhill-oddball commented:
Hey, @gunsch, @rroueche, @mnorthuis any input regarding the merging of this QA documentation into master?  Should I seek out anyone else's input instead / in addition?

mnorthuis commented:
Things look fine to me from a placement perspective.

gunsch commented on platform/quality-assurance/auth-session-loadtest.md:
```diff
@@ -0,0 +1,36 @@
+
+## Things you need
+1. [SOCKS proxy access](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/internal-tools/)
+2. [Access to internal tools](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/internal-tools/) (specifically Jenkins, but if you have access to one then you tend to have access to all)
```
fyi: in Markdown, you can use "1." for all your numbers and it'll auto-number them correctly (easier for editing in the future, don't have to change all the numbers if you insert/delete a step in the middle)

peterhill-oddball replied:
Excellent -- I was just wondering if that was best practice for MD yesterday.

gunsch commented on platform/quality-assurance/auth-session-loadtest.md:
```diff
@@ -0,0 +1,36 @@
+# Using authenticated sessions for load testing
+
+## Things you need
+1. [SOCKS proxy access](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/internal-tools/)
```
links for #1 and #2 are out of date, see
https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/onboarding/request-access-to-tools.md

peterhill-oddball replied:
Fixed

gunsch commented on platform/quality-assurance/auth-session-loadtest.md:
```diff
@@ -0,0 +1,36 @@
+## Things you need
+1. [SOCKS proxy access](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/internal-tools/)
+2. [Access to internal tools](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/internal-tools/) (specifically Jenkins, but if you have access to one then you tend to have access to all)
+3. Access to [Devops](https://github.com/department-of-veterans-affairs/devops) repository
```
```suggestion
3. Access to [`devops`](https://github.com/department-of-veterans-affairs/devops) repository
```

gunsch commented on platform/quality-assurance/auth-session-loadtest.md:
```diff
@@ -0,0 +1,36 @@
+
+## Steps to create authenticated sessions
+1. Start up the SOCKS proxy
+2. Navigate to [this](http://jenkins.vfs.va.gov/job/rake_tasks/job/vets-api-load-test-sessions/) Jenkins task in a browser
```
```suggestion
2. Navigate to the [vets-api-load-test-sessions](http://jenkins.vfs.va.gov/job/rake_tasks/job/vets-api-load-test-sessions/) Jenkins task in a browser
```

gunsch commented on platform/quality-assurance/auth-session-loadtest.md:
```diff
@@ -0,0 +1,36 @@
+11. Use the array of sessions in `sessions.json` with Python's [json library](https://docs.python.org/3/library/json.html)
+12. Run your load test
+
+[jenkins-login]: https://github.com/department-of-veterans-affairs/va.gov-team/blob/peter-qa/platform/quality-assurance/images/jenkins-login.png
```
these are all referencing the `peter-qa` branch, can you make them relative links instead so they reference current branch? (here and throughout the PR)

peterhill-oddball replied:
Will do

gunsch commented on platform/quality-assurance/create-an-issue.md:
```diff
@@ -0,0 +1,20 @@
+# How to Create an Issue
+
+1. Navigate to relevant repo in GitHub.  Example: [vets-website](https://github.com/department-of-veterans-affairs/vets-website/issues)
+2. Click New Issue
```
```suggestion
2. Click "New Issue"
```

gunsch commented on platform/quality-assurance/create-an-issue.md:
```diff
@@ -0,0 +1,20 @@
+      - Error messages
+      - Stack trace
+    - Expected behavior
+4. Assign applicable labels.  Consider labels for:
```
```suggestion
4. Assign applicable labels.  Include labels for:
```

(are these labels optional or required?)

gunsch commented on platform/quality-assurance/create-an-issue.md:
```diff
@@ -0,0 +1,20 @@
+
+1. Navigate to relevant repo in GitHub.  Example: [vets-website](https://github.com/department-of-veterans-affairs/vets-website/issues)
+2. Click New Issue
+3. Include relevant details required to reproduce the issue
```
can we put  all these details  into the issue template, rather than in the documentation? would prefer putting it closer to where the issue gets filled out / harder to miss

peterhill-oddball replied:
That makes sense, and the thought definitely occurred to me.  I wasn't sure what the process is for adding a template for use in GH.  I can make the template creation a to-do for this sprint.

gunsch commented on platform/quality-assurance/deliverables.md:
```diff
@@ -0,0 +1,21 @@
+    - Example: [Test Execution Example](https://drive.google.com/file/d/1_DeDS8lE2yxjcYPd_TEiSANl65Fsn95s/view?usp=sharing)
+4. Requirements Traceability Matrix (RTM)
+    - This artifact usually takes the form of a Word / Google Doc. The RTM artifact is a mapping from requirement / user story to test case that demonstrates how the functional changes in your product were tested for this release.
+    - Example: [RTM Template](https://drive.google.com/file/d/1fAWBuscDAGLLC18C5lbekjvjGElAaAm4/view?usp=sharing)
```
can you confirm this file (and any other Drive files linked here) are in the VSP Team drive?

peterhill-oddball replied:
They are now!

gunsch reviewed (commented):
handful of comments (mostly small editing tweaks), thanks for putting this together!

pjhill committed:
Update platform/quality-assurance/create-an-issue.md

Co-Authored-By: Andrew Gunsch <andrew.gunsch@adhocteam.us>

pjhill committed:
Update platform/quality-assurance/create-an-issue.md

Co-Authored-By: Andrew Gunsch <andrew.gunsch@adhocteam.us>

pjhill committed:
Update platform/quality-assurance/auth-session-loadtest.md

Co-Authored-By: Andrew Gunsch <andrew.gunsch@adhocteam.us>

pjhill committed:
Update platform/quality-assurance/auth-session-loadtest.md

Co-Authored-By: Andrew Gunsch <andrew.gunsch@adhocteam.us>

pjhill committed:
Update deliverables.md

pjhill committed:
Update auth-session-loadtest.md

- Adjusting links for internal tools to updated documentation
- Switching to MD best practice of using "1." for all numbered list items

pjhill committed:
Update auth-session-loadtest.md

Switched to relative links for image files

pjhill committed:
Update auth-session-loadtest.md
